tool/dart_model_generator/lib/generate_dart_model.dart

```diff
@@ -37,16 +37,45 @@ String generate(String schemaJson,
   final schema = JsonSchema.create(schemaJson,
       refProvider: LocalRefProvider(dartModelJson ??
           File('schemas/dart_model.schema.json').readAsStringSync()));
-  for (final def in schema.defs.entries) {
-    if (def.value.oneOf.isNotEmpty) {
-      result.add(_generateUnion(def.key, def.value.oneOf));
+  final allDefinitions = <String, JsonSchema>{
+    for (final def in schema.defs.entries) def.key: def.value,
+  };
+
+  for (final MapEntry(:key, :value) in allDefinitions.entries) {
+    if (value.oneOf.isNotEmpty) {
```

Comment:
I just merged another PR tweaking generation for `oneOf`, because the schema was not correct, please rebase; sorry about that. It moves the `oneOf` in the schema into a property called `value`, since that's what we write on the wire.

https://github.com/dart-lang/macros/pull/26